packages/manager/modules/pci/src/projects/project/instances/add/add.controller.js

```diff
@@ -847,16 +847,38 @@ export default class PciInstancesAddController {
     if (!this.isLinuxImageType()) {
       this.instance.userData = null;
     }
-
-    if (this.isLocalPrivateMode() && this.isAttachPublicNetwork) {
-      this.instance.networks = [
-        ...this.instance.networks,
-        {
-          networkId: this.publicNetwork?.find(
-            (network) => network.name === PUBLIC_NETWORK,
-          )?.id,
+    if (this.isLocalPrivateMode()) {
+      const publicNetworkAlreadyExist = this.instance.networks.filter(
+        (instanceNetwork) => {
+          return this.publicNetwork?.find(
+            (network) => network.id === instanceNetwork.networkId,
+          );
         },
```

Comment:
**nitpick:** `Array.prototype.filter` is not a good choice if you're looking for a boolean. And as it can be seen in the code, you're depending on the length of the array to decide. You can replace `filter` with `some`.